### ISSUE TITLE
`redundant-object` complains about object self reuse in the nested formations

### ISSUE THREAD

**Issue by h1alexbel** [objectionary/lints#639]
Consider such EO snippet:

```eo
[] > jeff
  $ > self
  [] > say-hello
    stdout self > @
```

Here, `self` was introduced on a purpose, in order to make `jeff` itself reusable in the `say-hello`. Thus, `self` is not redundant. However, `redundant-object` complains as such:

```text
The object "self" is redundant, consider inline it instead (redundant-object (Single))
```

We should not complain in this case, I believe

**Comment by h1alexbel** [comment]
@yegor256 WDYT?

**Comment by h1alexbel** [comment]
@yegor256 ^ reminder

**Comment by bibonix** [comment]
@yegor256 I'm planning to work on this. The lint flags `self` in the example as redundant even though it's referenced inside the nested `say-hello` formation. I'll trace through the XSL's usage XPath and adjust it so that references from nested formations (i.e., reaching out through additional `ξ.ρ` hops) are counted as usages.

**Comment by bibonix** [comment]
Submitted #853 with the fix and a regression test (`allows-self-reused-in-nested-formation.yaml`). The lint was matching the `ξ.ρ.self` usage correctly, but its remediation (inline) is unsafe when the candidate's body is `$` and the usage sits in a nested formation, because `ξ` there means something different. The fix suppresses the warning only for that specific shape; same-level `$` aliases are still flagged as redundant.

**Comment by volodya-lombrozo** [comment]
This one was fixed in the #639
